esp-bootloader-esp-idf/src/lib.rs

```diff
@@ -9,6 +9,69 @@
 //! - read the partition table
 //! - conveniently use a partition to read and write flash contents
 //!
+//! ## Examples
+//!
+//! ### Populating the Application Descriptor
+//!
+//! To use the default values:
+//!
+//! ```rust, no_run
+//! #![no_std]
+//! #![no_main]
+//!
+//! #[panic_handler]
+//! fn panic(_: &core::panic::PanicInfo) -> ! {
+//!     loop {}
+//! }
+//!
+//! esp_bootloader_esp_idf::esp_app_desc!();
+//!
+//! #[esp_hal::main]
+//! fn main() -> ! {
+//!     let _peripherals = esp_hal::init(esp_hal::Config::default());
+//!
+//!     loop {}
+//! }
+//! ```
+//!
+//! If you want to customize the application descriptor:
+//!
+//! ```rust, no_run
+//! #![no_std]
+//! #![no_main]
+//!
+//! #[panic_handler]
+//! fn panic(_: &core::panic::PanicInfo) -> ! {
+//!     loop {}
+//! }
+//!
+//! esp_bootloader_esp_idf::esp_app_desc!(
+//!     // Version
+//!     "1.0.0",
+//!     // Project name
+//!     "my_project",
+//!     // Build time
+//!     "12:00:00",
+//!     // Build date
+//!     "2021-01-01",
+//!     // ESP-IDF version
+//!     "4.4",
+//!     // MMU page size
+//!     8 * 1024,
+//!     // Minimal eFuse block revision supported by image
```

Comment:
maybe adding "Format: major * 100 + minor" here and for the maximum might be useful